parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -445,6 +600,26 @@ public FieldEnumConverter(List<TProtocol> events, ThriftField field) {
       }
     }
 
+     @Override
+     protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
+       Binary value = d.decodeToBinary(i);
+       final Integer id = enumLookup.get(value);
```

Comment:
There's quite a bit of code duplication as well, which could be cleaned up. One of the problems with the code duplication is that it creates multiple anonymous classes when one would be sufficient. It isn't necessary to add them as named classes because cutting down on the code duplication should fix it.